src/engine/strat_engine/setup.rs

```diff
@@ -198,7 +198,18 @@ pub fn get_blockdevs(pool_uuid: PoolUuid,
                     RangeAllocator::new(actual_size,
                                         segment_table.get(bda.dev_uuid()).unwrap_or(&vec![]))?;
 
-                blockdevs.push(StratBlockDev::new(*device, devnode.to_owned(), bda, allocator));
+                let (location, disk_id) = pool_save
+                    .block_devs
+                    .get(bda.dev_uuid())
+                    .map(|bd_save| (bd_save.user_info.clone(), bd_save.hardware_info.clone()))
```

Comment:
This should be an error, not an ```unwrap_or()```, because it would mean there is no entry for the blockdev in the pools metadata.